lib/execution_engine2/utils/SDKMethodRunner.py

```diff
@@ -706,3 +706,52 @@ def start_job(self, job_id, ctx, skip_estimation=True):
 
         with self.get_mongo_util().mongo_engine_connection():
             job.save()
+
+    def check_job(self, job_id, ctx):
+        """
+        check_job: check and return job status for a given job_id
+
+        Parameters:
+        job_id: id of job
+        """
+        logging.info("Start fetching status for job: {}".format(job_id))
+
+        if not job_id:
+            raise ValueError("Please provide valid job_id")
+
+        self.check_permission_for_job(job_id=job_id, ctx=ctx, write=False)
+
+        job_state = self.get_mongo_util().get_job(job_id=job_id).to_mongo().to_dict()
+
+        return job_state
+
+    def check_jobs(self, job_ids, ctx):
+        """
+        check_jobs: check and return job status for a given of list job_ids
+
+        """
+
+        logging.info("Start fetching status for jobs: {}".format(job_ids))
+
+        job_states = dict()
+
+        for job_id in job_ids:
+            job_states[job_id] = self.check_job(job_id, ctx)
```

Comment:
I think we should make make check_jobs a private function and then do a manual mongoengine query  select job where job in [id1,id2,id3] 
I don't think we need to check permission for each job, because if you have permission to the workspace, you should have permission to all the jobs.

Reply:
👍 I added a check_permission params to by pass this check when calling from `check_workspace_jobs`